codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java

```diff
@@ -276,6 +277,32 @@ public void dataReadForUnknownStreamShouldApplyFlowControlAndFail() throws Excep
         }
     }
 
+    @Test(expected = Http2Exception.class)
+    public void dataReadForUnknownStreamThatCouldntExistFail() throws Exception {
+        when(connection.streamMayHaveExisted(STREAM_ID)).thenReturn(false);
+        when(connection.stream(STREAM_ID)).thenReturn(null);
+        final ByteBuf data = dummyData();
+        int padding = 10;
+        int processedBytes = data.readableBytes() + padding;
+        try {
+            decode().onDataRead(ctx, STREAM_ID, data, padding, true);
+        } catch (Http2Exception ex) {
+            assertFalse(ex instanceof Http2Exception.StreamException);
```

Comment:
nit: consider using `assertThat(ex, not(instanceOf(Http2Exception.StreamException.class)));`